.gitignore

```diff
@@ -0,0 +1,9 @@
+# Ignore backup files created by editors like emacs.
+*~
+# Ignore cmake files.
+*CMakeCache.txt
+*CMakeFiles
+# Ignore tmp directory/file. We use it sometimes to write debug information.
+tmp
+# Ignore build trees.
+build-*
```

Comment:
Does the hyphen here mean that directories called `build` won't get ignored? I think those are fairly common.